src/shared/components/confirm_deletion.js

```diff
@@ -26,7 +26,12 @@ module.exports = React.createClass({
   },
 
   which_text: function () {
-    return this.state.confirmation ? 'Confirm' : `${this.props.payment ? 'X' : 'Revoke Gift Aid' }`
+    const button_text =
```

Comment:
This should be passed in, the deletion button shouldn't need to know about all the ways it's used in the app